File: problems/coding/015-longest-substr-no-repeat/solution.py
def lengthOfLongestSubstring(s):
    seen = {}; best = 0; left = 0
    for i, c in enumerate(s):
        if c in seen and seen[c] >= left: left = seen[c] + 1
        seen[c] = i; best = max(best, i - left + 1)
    return best

File: problems/coding/015-longest-substr-no-repeat/solution.js
function lengthOfLongestSubstring(s) {
  const seen = new Map(); let best = 0, left = 0;
  for (let i = 0; i < s.length; i++) {
    const c = s[i];
    if (seen.has(c) && seen.get(c) >= left) left = seen.get(c) + 1;
    seen.set(c, i); best = Math.max(best, i - left + 1);
  }
  return best;
}
